ReactAndroid/src/main/java/com/facebook/react/uimanager/UIManagerModule.java

```diff
@@ -85,6 +89,17 @@ public UIManagerModule(
     super(reactContext);
     mEventDispatcher = new EventDispatcher(reactContext);
     DisplayMetrics displayMetrics = reactContext.getResources().getDisplayMetrics();
+
+    // Get the real display metrics if we are using API level 17 or higher.
+    // The real metrics include system decor elements (e.g. soft menu bar).
+    //
+    // See: http://developer.android.com/reference/android/view/Display.html#getRealMetrics(android.util.DisplayMetrics)
+    if (Build.VERSION.SDK_INT >= 17) {
+      WindowManager wm = (WindowManager)reactContext.getApplicationContext().getSystemService(Context.WINDOW_SERVICE);
+      Display display = wm.getDefaultDisplay();
+      display.getRealMetrics(displayMetrics);
+    }
```

Comment:
If any of this fails (`getSystemService` returns null, the result is not of type `WindowManager` etc.) we probably don't want to crash the whole `UIManagerModule`. Can you wrap all of this in a try-catch block and log the exception using `FLog` (check other call sites of `FLog` for examples).